packages/menu/src/Menu.ts

```diff
@@ -125,6 +198,77 @@ export class Menu extends SpectrumElement {
         this.removeEventListener('keydown', this.handleKeydown);
     }
 
+    public async selectOrToggleItem(item: MenuItem): Promise<void> {
+        const resolvedSelects = this.resolvedSelects;
+        if (resolvedSelects == null) {
+            return;
+        }
+
+        const oldSelectedItemsMap = new Map(this.selectedItemsMap);
+        const oldSelected = this.selected.slice();
+        const oldSelectedItems = this.selectedItems.slice();
+        const oldValue = this.value;
+
+        if (resolvedSelects === 'single') {
+            this.selectedItemsMap.clear();
+            this.selectedItemsMap.set(item, true);
+            this.value = item.value;
+            this.selected = [item.value];
+            this.selectedItems = [item];
+        } else {
+            if (this.selectedItemsMap.has(item)) {
+                this.selectedItemsMap.delete(item);
+            } else {
+                this.selectedItemsMap.set(item, true);
+            }
+
+            // Match HTML select and set the first selected
+            // item as the value. Also set the selected array
+            // in the order of the menu items.
+            let valueSet: boolean = false;
+            let selected: string[] = [];
+            let selectedItems: MenuItem[] = [];
```

Comment:
I wonder why you linter doesn't complain at these two values not being `const`?

Reply:
I'll need to see what about mine does that.

Reply:
that's weird. I'd expect the linter to have at least caught this at commit time.